interop/tapir/src/main/scala/caliban/interop/tapir/TapirAdapter.scala

```diff
@@ -199,6 +217,37 @@ object TapirAdapter {
     )
   }
 
+  private def encodeTextEventStreamResponse[E, BS](
+    resp: GraphQLResponse[E]
+  )(implicit streamConstructor: StreamConstructor[BS], responseCodec: JsonCodec[ResponseValue]): CalibanBody[BS] = {
+    val response: ZStream[Any, Throwable, ServerSentEvent] = resp.data match {
+      case ObjectValue(fields) =>
+        fields.foldLeft(ZStream.empty: ZStream[Any, Throwable, ServerSentEvent]) { case (_, v) =>
+          v match {
+            case (fieldName, StreamValue(stream)) =>
+              stream.map { r =>
+                ServerSentEvent(
+                  Some(
+                    responseCodec.encode(
+                      GraphQLResponse(
+                        ObjectValue(List(fieldName -> r)),
+                        resp.errors
+                      ).toResponseValue
+                    )
+                  ),
+                  Some("next")
+                )
+              }.orDie
```

Comment:
Is the `.orDie` necessary since the method return type is `ZStream[Any, Throwable, ServerSentEvent]`?